### ISSUE TITLE
`@graphql-eslint/no-unreachable-types` reports issues with interfaces

### ISSUE THREAD

**Issue by dotansimha** [dotansimha/graphql-eslint#315]
Given this

```
interface User {

}

type SuperUser implements User {

}

type Query {
  user: User!
}
```

We'll get an error regarding `SuperUser` no in use.

**Comment by dnalborczyk** [comment]
@dotansimha seems to be correct. `SuperUser` is not connected to the root. I thought that would be the whole point of this linting rule?

**Comment by dotansimha** [comment]
@dnalborczyk but it does implementing the `User`. And GraphQL schema can declare the interface as return type, and the actual type is defined during resolvers execution. @kamilkisiela what do you think?

**Comment by kamilkisiela** [comment]
Yeah, I need to fix it.

**Comment by dotansimha** [comment]
Fixed in `@graphql-eslint/eslint-plugin@0.8.1`

**Comment by ostrowr** [comment]
I can open another issue if helpful, but I think this one should be reopened. The linked PR solves the issue for types that are only reachable directly from an interface, but not for one that is available in an implementation of the interface.

Using the example above:

```
interface User {
    id: String
}

type SuperUser implements User {
    id: String 
    superDetail: SuperDetail
}

# This one is still marked as unreachable!
type SuperDetail {
    detail: String
}

type Query {
  user: User!
}
```

Additionally, 

`extend type SuperUser...`

is valid GraphQL but will also, strangely, be marked as unreachable even though SuperUser is marked as reachable.

**Comment by kamilkisiela** [comment]
I'll fix it

**Comment by kamilkisiela** [comment]
#361

**Comment by dotansimha** [comment]
Fixed in `@graphql-eslint/eslint-plugin@0.9.1`. Thanks @kamilkisiela !
